tests/feats/utils.py

```diff
@@ -0,0 +1,126 @@
+from unittest import TestCase
+
+from feats.meta import Implementation
+from feats.utils import fn_to_implementations
+from feats.utils import obj_to_implementations
+
+
+class FullySpecified:
+    def nullary(self) -> str:
+        return "nullary"
+
+    def unary(self, arg1: str) -> int:
+        return int(arg1)
+    unary._feats_annotations_ = ["special"]
+
+    def binary(self, arg1: str, arg2: int) -> float:
+        return float(arg2)
+    binary._feats_annotations_ = ["special", "default"]
+
+    def _private(self) -> str:
+        return "Don't look at me!"
+    _private._feats_annotations_ = ["hidden"]
+
+
+class Incomplete:
+    def no_return(self):
+        return "no_return"
+
+    def no_input_type(self, arg1) -> int:
+        return "no_input_type"
+
+    def partial_input_type(self, arg1, arg2: str) -> str:
+        return "partial_input_type"
+
+
+class Empty:
+    pass
+
+
+def FullySpecifiedFunc(arg: str) -> bool:
+    return True
+
+
+FullySpecifiedFunc._feats_annotations_ = ["default"]
```

Comment:
To prevent leaking the internals of feats, we can use the @default annotation here.